### PR TITLE
[doc] Document the `--flex-alloc-tasks` feature

### LINKED ISSUE #547: Document the '--flex-alloc-tasks' cmd option
* Document the flexible allocation of the number of tasks. 

* Create a tutorial example demonstrating the use of `--flex-alloc-tasks` command line option.

### PR DESCRIPTION
**Pull request by teojgo**
* Add a tutorial example.

Closes #547

### CODE REVIEW

File: docs/running.rst [hunk 1]
```diff
@@ -1000,19 +1000,32 @@ If you now try to run a test that loads the module `cudatoolkit`, the following
...
-For both ``idle`` and ``all`` the final number of tasks will be the product of the nodes satisfying the required criteria and :attr:`num_tasks_per_node <reframe.core.pipeline.RegressionTest.num_tasks_per_node>`.
-Finally, by specifying a positive integer ReFrame is going to use that number as the number of tasks without doing any additional check on the nodes.
+ReFrame can automatically set the number of tasks of a particular test, if its :attr:`num_tasks <reframe.core.pipeline.RegressionTest.num_tasks>` attribute is set to ``0``.
+By default, ReFrame will spawn such a test on all the idle nodes of the current system partition.
+This behavior can be adjusted using the ``--flex-alloc-tasks`` command line option.
+This option accepts three values:
+
+  1. ``idle``: (default) In this case, ReFrame will set the number of tasks to the number of idle nodes of the current logical partition.
```
Comment (teojgo) [inline docs/running.rst]: The number of tasks equals the number of nodes multiplied by the number of tasks per node. Therefore, it should  be mentioned here.

File: docs/running.rst [hunk 2]
```diff
@@ -1000,19 +1000,32 @@ If you now try to run a test that loads the module `cudatoolkit`, the following
...
+  2. ``all``: In this case, ReFrame will set the number of tasks to the number of all the nodes of the current logical partition.
```
Comment (teojgo) [inline docs/running.rst]: Same as above.

File: docs/running.rst (hunk 1 above)
Comment (vkarak) [reply docs/running.rst]: Can you fix that?

File: docs/running.rst (hunk 2 above)
Comment (vkarak) [reply docs/running.rst]: Can you fix that?

**Pull request merged**